cJSON.c

```diff
@@ -452,20 +431,22 @@ static cJSON_bool print_number(const cJSON * const item, printbuffer * const out
     {
         length = sprintf((char*)number_buffer, "null");
     }
-    else if ((fabs(floor(d) - d) <= DBL_EPSILON) && (fabs(d) < 1.0e60))
```

Comment:
I think it still makes sense to print integers without engineering notation, as long as they can be printed with full precision.

So instead of removing this integer handling completely, 1e60 should be replaced with 2^53.

I'm just not sure if this integer detection with `DBL_EPSILON` is correct, this might have to be fixed.